cmd/fetch_repo/module.go

```diff
@@ -93,9 +94,11 @@ func fetchModule(dest, importpath, version, sum string) error {
 	// Parse the JSON output.
 	var dl struct{ Dir, Sum, Error string }
 	if err := json.Unmarshal(buf.Bytes(), &dl); err != nil {
+		_, _ = os.Stderr.Write(bufErr.Bytes())
```

Comment:
This function shouldn't directly write to stdout or stderr; we should avoid doing that above with `dlErr` (no need to make that change in this CL though).

Instead, `err` should contain the error text. For this case:

```
if err := json.Unmarshal(buf.Bytes(), &dl); err != nil {
  if bufErr.Len() > 0 {
    return fmt.Errorf("%s %s: %s", cmd.Path, strings.Join(cmd.Args, " "), bufErr.Bytes())
  } else {
    return fmt.Errorf("%s %s: %v", cmd.Path, strings.Join(cmd.Args, " "), err)
  }
}
```